helmclient.go

```diff
@@ -394,7 +394,16 @@ func (c *Client) InstallFromTarball(path, ns string, options ...helmclient.Insta
 		release, err := c.newHelmClientFromTunnel(t).InstallRelease(path, ns, options...)
 		if IsCannotReuseRelease(err) {
 			return backoff.Permanent(microerror.Mask(err))
+		} else if IsReleaseAlreadyExists(err) {
+			return backoff.Permanent(microerror.Mask(err))
+		} else if IsTarballNotFound(err) {
+			return backoff.Permanent(microerror.Mask(err))
 		} else if IsReleaseNotFound(err) {
+			// This is out of alphabetical order. But
+			// IsReleaseNotFound is rather bold And I want to catch
+			// other errors first. I don't know why do we even
+			// match by it here. Also it would be good to have
+			// tests of all the matchers.
```

Comment:
As I wrote here. I can't tell how this error can happen during install. Also matching isn't very accurate IMO.
If there was a case for this error I would write a test.

Reply:
The method is called `InstallRelease`. So I assume when you try to install a release that does not exist an "release not found" error is returned. That sounds pretty legit to me

Reply:
The release must not exist in order to install it. Otherwise `IsReleaseAlreadyExists` I just introduced is returned.
If you speak about the chart tarball then I added `IsTarballNotFound`.

Still I don't get how it can return `IsReleaseNotFound`.

Reply:
Also a test for `IsTarballNotFound` is written below.

Reply:
Oh wow, yeah. Totally. Maybe that was just bad copy/paste. Lets drop it then, because this does not make any sense and when the installation is not returning such a specific error then this is useless.

Reply:
I forgot to remove. https://github.com/giantswarm/helmclient/pull/54.